zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/usertask/UserTaskCommandProcessors.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.engine.processing.usertask;
+
+import io.camunda.zeebe.engine.processing.bpmn.behavior.BpmnBehaviors;
+import io.camunda.zeebe.engine.processing.common.EventHandle;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.Writers;
+import io.camunda.zeebe.engine.processing.usertask.processors.UserTaskAssignProcessor;
+import io.camunda.zeebe.engine.processing.usertask.processors.UserTaskClaimProcessor;
+import io.camunda.zeebe.engine.processing.usertask.processors.UserTaskCommandProcessor;
+import io.camunda.zeebe.engine.processing.usertask.processors.UserTaskCompleteProcessor;
+import io.camunda.zeebe.engine.processing.usertask.processors.UserTaskUpdateProcessor;
+import io.camunda.zeebe.engine.state.mutable.MutableProcessingState;
+import io.camunda.zeebe.protocol.record.intent.UserTaskIntent;
+import java.util.EnumMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public final class UserTaskCommandProcessors {
+
+  private final Map<UserTaskIntent, UserTaskCommandProcessor> commandToProcessor;
+
+  public UserTaskCommandProcessors(
+      final MutableProcessingState processingState,
+      final BpmnBehaviors bpmnBehaviors,
+      final Writers writers) {
+    final var keyGenerator = processingState.getKeyGenerator();
+    final EventHandle eventHandle =
+        new EventHandle(
+            keyGenerator,
+            processingState.getEventScopeInstanceState(),
+            writers,
+            processingState.getProcessState(),
+            bpmnBehaviors.eventTriggerBehavior(),
+            bpmnBehaviors.stateBehavior());
+
+    commandToProcessor =
+        new EnumMap<>(
+            Map.of(
+                UserTaskIntent.ASSIGN,
+                new UserTaskAssignProcessor(processingState, writers),
+                UserTaskIntent.CLAIM,
+                new UserTaskClaimProcessor(processingState, writers),
+                UserTaskIntent.UPDATE,
+                new UserTaskUpdateProcessor(processingState, writers),
+                UserTaskIntent.COMPLETE,
+                new UserTaskCompleteProcessor(processingState, eventHandle, writers)));
```

Comment:
Thank you, @korthout, for your suggestion!
I agree that it would simplify the code for this refactoring PR.

However, looking ahead to when we introduce `COMPLETE_TASK_LISTENER` command, we'll need to dynamically find the appropriate command processor to finalize the original command after all task listeners have been executed. This will require to have some kind of `UserTaskCommandProcessor` "registry". 

For this reason, I'd rather keep this class, so we can conveniently access all `UserTaskCommandProcessor` instances from the `UserTaskProcessor`.

Thanks again for your input! Happy to hear your thoughts on this.